openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/DataProductRepository.java

```diff
@@ -114,10 +116,14 @@ private void updateAssets() {
       // Remove assets that were deleted
       for (EntityReference asset : deleted) {
         deleteRelationship(original.getId(), DATA_PRODUCT, asset.getId(), asset.getType(), Relationship.HAS);
+        EntityInterface entityInterface = Entity.getEntity(asset, "*", Include.ALL);
+        searchRepository.updateEntity(entityInterface);
```

Comment:
why are you calling this inline, why can't we do this in post updates

Reply:
Pushed the change, @karanh37 can you please verify from you end is it working as expected ?

Reply:
Verified on UI, changes are working fine for Data Product.